From 49893becc0f82bdb4b5e3bd825bc2a027c7a7eec Mon Sep 17 00:00:00 2001
From: "Owen W. Taylor" <otaylor@fishsoup.net>
Date: Mon, 13 Feb 2017 01:20:43 +0000
Subject: [PATCH] Add a --llvm-config-dir option

The --llvm-prefix option doesn't quite work when building against
a yocto base, because the llvm-config-binary is in
$bindir/llvm3.3/<targetname>-llvm-config, which is not $prefix/bin
for any prefix.
---
 configure.ac | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 121457bdfd..753fe98f6f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -843,6 +843,12 @@ AC_ARG_WITH([llvm-prefix],
     [llvm_prefix="$withval"],
     [llvm_prefix=''])
 
+AC_ARG_WITH([llvm-config-dir],
+    [AS_HELP_STRING([--with-llvm-config-dir],
+        [Directory where llvm-config binary lives])],
+    [llvm_config_dir="$withval"],
+    [llvm_config_dir=''])
+
 PKG_CHECK_MODULES([LIBELF], [libelf], [have_libelf=yes], [have_libelf=no])
 if test "x$have_libelf" = xno; then
    LIBELF_LIBS=''
@@ -853,8 +859,11 @@ if test "x$have_libelf" = xno; then
 fi
 
 if test -z "$LLVM_CONFIG"; then
-    if test -n "$llvm_prefix"; then
-        AC_PATH_TOOL([LLVM_CONFIG], [llvm-config], [no], ["$llvm_prefix/bin"])
+    if test -n "$llvm_prefix" -a -z "$llvm_config_dir" ; then
+       llvm_config_dir="$llvm_prefix/bin"
+    fi
+    if test -n "$llvm_config_dir"; then echo ============== $llvm_config_dir/${ac_tool_prefix}llvm-config
+        AC_PATH_TOOL([LLVM_CONFIG], [llvm-config], [no], ["$llvm_config_dir"])
     else
         AC_PATH_TOOL([LLVM_CONFIG], [llvm-config], [no])
     fi
-- 
2.11.1

